src/common/tiny_vector.hpp

```diff
@@ -0,0 +1,144 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*- 
+// vim: ts=8 sw=2 smarttab
+//
+// Ceph - scalable distributed file system
+//
+// Copyright (C) 2004-2006 Sage Weil <sage@newdream.net>
+//
+// This is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License version 2.1, as published by the Free Software
+// Foundation.  See file COPYING.
+//
+
+#ifndef CEPH_COMMON_TINY_VECTOR_H
+#define CEPH_COMMON_TINY_VECTOR_H
+
+#include <type_traits>
+
+namespace ceph {
+
+// tiny_vector - a CPU-friendly container for non-movable objects.
```

Comment:
FIXME: add the slogan: *It's like small_vector but for mutexes, atomics and other non-movable things.*